clang/lib/Sema/SemaExprCXX.cpp

```diff
@@ -6719,6 +6719,16 @@ QualType Sema::CheckVectorConditionalTypes(ExprResult &Cond, ExprResult &LHS,
             : UsualArithmeticConversions(LHS, RHS, QuestionLoc,
                                          ACK_Conditional);
 
+    if (ResultElementTy.isNull()) {
+      Diag(QuestionLoc, diag::err_conditional_vector_mismatched)
+          << LHSType << RHSType;
+      return {};
+    }
```

Comment:
So we are checking if we are dealing with the same type just above.

If we don't support different types, then the call to `UsualArithmeticConversions` is useless, so it can be removed and this can be further simplified:

```suggestion
    if (!Context.hasSameType(LHSType, RHSType)) {
      Diag(QuestionLoc, diag::err_conditional_vector_mismatched)
          << LHSType << RHSType;
      return {};
    }
    QualType ResultElementTy = Context.getCommonSugaredType(LHSType, RHSType);
```

Reply:
Yeah, it looks better, thanks! 

I am actually not sure if we support different types or not. `vector_size` is an extension and there is no clear wording what is allowed and what is not...